### ISSUE TITLE
Research and prototype an update to states guidance on Design System website

### ISSUE THREAD

**Issue by CharlotteDowns** [alphagov/govuk-design-system#2593]
## What
[We identified gaps in the Design System styles guidance](https://docs.google.com/document/d/1jMKkh0Qjr17Cam-fWIyVULTHF2sEC64UG6uhd5aZunU/edit?usp=sharing). This exploration is specifically focused on drafting new guidance around 'states' and will replace [Understanding focus state styles](https://design-system.service.gov.uk/get-started/focus-states/).

## Why
Writing this guidance will probably force us to face quite a few inconsistencies across different components…

This will be used as reference for users creating new components. 

Outline:

- focus states
  - Charlotte D started work to [synthesise and determine guidelines on focus states](https://github.com/alphagov/govuk-design-system/issues/1853) – i.e. how they look for links, buttons, images, banners, larger areas
- hover states
  - link underlines
  - using tints on buttons (and clickable things with larger touch areas like tabs)
- active states
  - basically the same as focus?
  - except buttons which do the skeuomorphic press-down thing by removing the bottom shadow
- ‘selected’ / current page

## Who needs to work on this
Designer

## Who needs to review this
The GOV.UK Design System team

## Done when
- [x] Carried out research on how other design systems approach states and lessons to learn from.
- [x] Pulled together other issues and pull requests that may be relevant to changes made.
- [x] Start a draft guidance a google doc.

Moving to new card:
- Draft a PR that includes new guidance and suggestions on where it could live.

**Comment by CharlotteDowns** [comment]
Working document for findings from research of other design systems

**Comment by CharlotteDowns** [comment]
Other related issues:
- [Add disabled styles for form controls](https://github.com/alphagov/govuk-frontend/pull/3187)
- [Disabled select text colour too faint?](https://github.com/alphagov/govuk-frontend/issues/1692)

Supporting 'in active'/disabled states gets complicated because [we strongly encourage teams not to use them](https://design-system.service.gov.uk/components/button/#disabled-buttons). However there may be contexts where user research shows that they help users in particular user journeys, in these scenarios we want to be able to allow teams to provide the best experience.
